snapcraft.yaml

```diff
@@ -61,6 +61,13 @@ parts:
       - python3-yaml
     override-pull: |
       craftctl set version="$(/bin/date +%Y%m%d)"
+      craftctl default
+    override-build: |
+      # apply fips patch if the remote is a fips remote
+      if git remote get-url origin | grep "fips"; then
+        git apply "$CRAFT_PROJECT_DIR"/snap-patch/01-switch-to-fips.patch
```

Comment:
I am not sure using a patch is the best thing. Because one has to update the patch any time a line it in the context is modified. Could we use an environment variable?

Also I feel like checking the remote should be done in override-pull, then write down the what mode to build in a file.